plugin/src/test/java/io/jenkins/plugins/casc/SchemaGenerationTest.java

```diff
@@ -0,0 +1,60 @@
+package io.jenkins.plugins.casc;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import io.jenkins.plugins.casc.misc.JenkinsConfiguredWithCodeRule;
+import io.jenkins.plugins.casc.misc.Util;
+import org.everit.json.schema.Schema;
+import org.everit.json.schema.ValidationException;
+import org.everit.json.schema.loader.SchemaLoader;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+import org.junit.Rule;
+import org.junit.Test;
+
+import static io.jenkins.plugins.casc.SchemaGeneration.generateSchema;
+import static org.junit.Assert.fail;
+
+public class SchemaGenerationTest {
```

Comment:
https://github.com/skyscreamer/JSONassert could help

Reply:
Generally I would suggest adding more examples and to start building a mini test framework already. Current implementation is not extensible

Reply:
Yeah I will add more examples to the test framework asap.